components/admin/lessons/AdminLessonInputs.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState } from 'react'
+import { useAddModuleMutation } from '../../../graphql'
+import { formChange } from '../../../helpers/formChange'
+import { FormCard, Option } from '../../FormCard'
+import { AlertFillIcon, CheckCircleIcon } from '@primer/octicons-react'
+import styles from '../../../scss/adminLessonInputs.module.scss'
+import { Spinner } from 'react-bootstrap'
+import { GraphQLError } from 'graphql-request/dist/types'
+
+type Props = {
+  lesson: { title: string; id: number }
+}
+
+enum Error {
+  INVALID_DATA = 'missing module name or description'
+}
+
+const values: Option[] = [
+  {
+    title: 'Module Name'
+  },
+  {
+    title: 'Description',
+    type: 'MD_INPUT'
+  }
+]
+
+const AdminLessonInputs = ({ lesson }: Props) => {
+  const [formOptions, setFormOptions] = useState(values)
+  const [name, content] = formOptions
+
+  const [addModuleMutation, { data, error, loading }] = useAddModuleMutation({
+    variables: {
+      content: (content.value as string) || '',
+      lessonId: lesson.id,
+      name: (name.value as string) || ''
+    }
+  })
+
+  const [errorMsg, setErrorMsg] = useState(error?.message || '')
+
+  const handleChange = async (value: string, propertyIndex: number) => {
+    await formChange(value, propertyIndex, formOptions, setFormOptions)
+  }
+
+  const OnSubmit = async () => {
+    try {
+      // Empty strings aren't handled in the resolver
+      if (!content.value || !name.value) {
+        return setErrorMsg(Error.INVALID_DATA)
+      }
+
+      setErrorMsg('')
+      await addModuleMutation()
+    } catch (err) {
+      setErrorMsg((err as GraphQLError).message)
+    }
+  }
+
+  const Message = () => {
```

Comment:
Can we change this variable name to contain more meaning? For example `NotificationMessage`, or anything that conveys more of what this component is doing.